core/modules/util/Error.h

```diff
@@ -74,6 +96,47 @@ class Error {
      * @return an output stream
      */
     friend std::ostream& operator<<(std::ostream &out, Error const& error);
+
+    int getCode() const {
+        return code;
+    }
+
+    void setCode(int code) {
+        this->code = code;
+    }
+
+    const std::string& getMsg() const {
+        return msg;
+    }
+
+    void setMsg(const std::string& msg) {
+        this->msg = msg;
+    }
+
+    int getStatus() const {
+        return status;
+    }
+
+    void setStatus(int status) {
+        this->status = status;
+    }
```

Comment:
I'd make all of set* method disappear, in my opinion that are not need and interface would be _much_ better without them.